soci/fbs/ztoc/FileMetadata.go

```diff
@@ -0,0 +1,283 @@
+// Code generated by the FlatBuffers compiler. DO NOT EDIT.
```

Comment:
Why are we checking in generated code? Why isn't this a 3 line Makefile change?

Reply:
It could be a Makefile change, but I generally prefer to have them in repos. So that I can see the effect of upgrading code generators.

Reply:
FYI containerd is having .pb.go files like https://github.com/containerd/containerd/tree/main/api

Reply:
> It could be a Makefile change, but I generally prefer to have them in repos. So that I can see the effect of upgrading code generators.

Then also check it in to a reference directory. I like that idea. It may be useful to some future git archeologist, or someone chasing a compatibility bug. We could even write a test to ensure that the _actual_ generated code matches the reference code.

But it is an anti-pattern to check in machine generated code and then actually build it into the product. Eventually, some chaotic neutral code-slinger is going to hand-modify it. And they'll probably screw it up in some subtle way. And even if they don't screw it up, we will have still taken on technical debt, because a future someone will have to figure out what to do when it _does_ come time to modify the schema.

Reply:
Sorry. I wasn't giving the full picture of the files above.

containerd/containerd and other contained packages usually

- Have auto-generated files committed due to the reasons above
- Have a GitHub Action to check these files are reproducible from the tools we expect (e.g. https://github.com/containerd/containerd/blob/f4095a6876c62dd4cd5e3749d2435b2daa9eaff8/.github/workflows/ci.yml#L105-L107)

That way, we can check the code changes due to tool changes and we can make sure nobody can manually edit these files.

Reply:
Added `install-flatc`, `flatc`, `check-flatc` actions in Makefile.

Reply:
`check-flatc` will become enabled in the next PR.

Reply:
Sounds good to me. Best of both worlds.